great_expectations/validator/validator.py

```diff
@@ -2010,7 +2027,7 @@ def __init__(
         if self.expectation_engine is None:
             from great_expectations.compatibility import pyspark
 
-            if pyspark.DataFrame and isinstance(batch.data, pyspark.DataFrame):
```

Comment:
@alexsherstinsky mypy was upset that we were evaluating this and it should always be truthy. Was this actually doing something?

Reply:
There is a little magic going on here. `pyspark` is imported on the line above from `great_expectations.compatibility`. If users don't have pyspark installed they will get an instance of our `NotImported` which overrides `__bool__` to be `False` (see [here](https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/compatibility/not_imported.py#L33)).
Checking `if pyspark.DataFrame` is basically a check whether pyspark is installed or not and we only want to do the `isinstance` check if it is installed.

Reply:
Gotcha. I had a hunch it was going to be some magic like that. I wonder if there's a better way that we can make what's going on here clear to mypy 🤔 . But obvs that's out of scope for this. I'll revert this part. Thanks!